src/containers/Home.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { fetchPetitions, clearPetitions } from 'actions/PetitionsActions';
+import Home from 'components/Home';
+import getPetitions from 'selectors/petitions';
+import getPetitionsHomeQuery from 'selectors/petitionsHomeQuery';
+import isClientSideRouting from 'helpers/isClientSideRouting';
+
+const HOME_QUERY = getPetitionsHomeQuery({ limit: 3 });
+
+const HomeContainer = withRouter(React.createClass({
+
+  componentWillMount () {
+    // If there are no petitions, or if the user arrived on the page by clicking
+    // a client-side router link, then we fetch petitions client-side.
+    if (!this.props.petitions.length || isClientSideRouting(this.props.location)) {
```

Comment:
Ya, having the separate state objects like `petitions.latest` and `petitions.trending` would simplify this. I'll take a shot at it